### PR TITLE
fix issue #3683 - fix ICU build on Linux

### PR DESCRIPTION
**Pull request by sbj42**
This addresses #3683.

The following changes allow ICU to build (and `find_package(ICU)` to succeed) on both my Linux and Windows build machines:
- use the ICU tgz distribution rather than the zip file, to avoid CRLF line endings on Linux
- don't use MSYS on Linux
- fix runConfigureICU options for Linux
- fix CFLAGS, CXXFLAGS, and LDFLAGS for Linux
- don't rename the resulting libraries on Linux

### CODE REVIEW

**Comment by dethtoll1** [comment]
Some positive feedback:

This issue was blocking my attempt to install pdal on Linux.  I was able to successfully install icu:x64-linux with the portfile in this PR.  Looking forward to seeing this integrated.

Now on to the next failing package...

**Comment by phongphan** [comment]
I have change this line 44 and 169 from
`if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")`
to 
`if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin")`
then everything works great. Hope this PR get merge soon.

File: ports/icu/portfile.cmake [hunk 1]
```diff
@@ -21,15 +21,60 @@ set(ICU_VERSION_MAJOR 61)
...
+else()
+    set(CONFIGURE_OPTIONS "${CONFIGURE_OPTIONS} --enable-static --disable-shared")
+endif()
+
+set(CONFIGURE_OPTIONS_RELASE "--disable-debug --enable-release --prefix=${CURRENT_PACKAGES_DIR}")
+set(CONFIGURE_OPTIONS_DEBUG  "--enable-debug --disable-release --prefix=${CURRENT_PACKAGES_DIR}/debug")
+
+if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
+
+set(BASH bash)
```
Comment (KindDragon) [inline ports/icu/portfile.cmake]: Can you add an indent for the code inside `if`?

File: ports/icu/portfile.cmake [hunk 2]
```diff
@@ -130,12 +166,15 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
     file(COPY ${RELEASE_DLLS} DESTINATION ${CURRENT_PACKAGES_DIR}/bin)
     file(COPY ${DEBUG_DLLS} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin)
 else()
-    # rename static libraries to match import libs
-    # see https://gitlab.kitware.com/cmake/cmake/issues/16617
-    foreach(MODULE dt in io tu uc)
-        file(RENAME ${CURRENT_PACKAGES_DIR}/lib/sicu${MODULE}.lib ${CURRENT_PACKAGES_DIR}/lib/icu${MODULE}.lib)
-        file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/sicu${MODULE}d.lib ${CURRENT_PACKAGES_DIR}/debug/lib/icu${MODULE}d.lib)
-    endforeach()
+    if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
Comment (KindDragon) [inline ports/icu/portfile.cmake]: if(NOT VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")

**Comment by ras0219-msft** [comment]
Thanks for the PR and sorry for the long delay!

**Pull request merged**